prow/git/v2/interactor.go

```diff
@@ -80,6 +82,33 @@ type cloner interface {
 	Clone(from string) error
 }
 
+// MergeOpt is a type of a function that returns an []string
+type MergeOpt func() []string
+
+func hasMessageOption(opts ...MergeOpt) bool {
```

Comment:
It would be easier at this point to just make `MergeOpt` a `struct` that has some fields like `CommitMessage` and then defaulting that field if unset and always doing a `-m mergeOpt.CommitMessage`

Reply:
The problem here is that we don't always want to default it. The commit message is only in the Merge strategy. However that can be defaulted only in that strategy

Reply:
The other strategies have a commit message as well or not? If its not implemented we should return an error at least. And having a proper struct rather than a string slice and then needing to parse the string slice seems much better to me

Reply:
+1 for the struct. It makes also sense to start validating the options, but the command will return error either way because of the wrong options. For example, in this case, if you pass a commit message option in a squash command.